x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/components/schema_callouts.test.tsx

```diff
@@ -0,0 +1,119 @@
+ * 2.0.
+ */
+
+import { setMockValues, setMockActions } from '../../../../__mocks__';
+import '../../../__mocks__/engine_logic.mock';
+
+import React from 'react';
+
+import { shallow } from 'enzyme';
+
+import { SchemaErrorsCallout } from '../../../../shared/schema';
+
+import {
+  UnsearchedFieldsCallout,
+  UnconfirmedFieldsCallout,
+  ConfirmSchemaButton,
+} from './schema_callouts';
+
+import { SchemaCallouts } from './';
+
+describe('SchemaCallouts', () => {
+  const values = {
+    hasUnconfirmedFields: false,
+    hasNewUnsearchedFields: false,
+    mostRecentIndexJob: {
+      hasErrors: false,
+      activeReindexJobId: 'some-id',
+    },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues(values);
+  });
+
+  it('renders nothing if there is nothing to call out', () => {
+    const wrapper = shallow(<SchemaCallouts />);
+
+    expect(wrapper.text()).toBeFalsy();
+  });
+
+  it('renders a schema errors callout if the most recent index job had errors', () => {
+    setMockValues({
+      ...values,
+      mostRecentIndexJob: {
+        hasErrors: true,
+        activeReindexJobId: '12345',
+      },
+    });
+    const wrapper = shallow(<SchemaCallouts />);
+
+    expect(wrapper.find(SchemaErrorsCallout)).toHaveLength(1);
+    expect(wrapper.find(SchemaErrorsCallout).prop('viewErrorsPath')).toEqual(
+      '/engines/some-engine/schema/reindex_job/12345'
+    );
+  });
+
+  it('renders an unsearched fields callout if the schema has new unconfirmed & unsearched fields', () => {
+    setMockValues({
+      ...values,
+      hasUnconfirmedFields: true,
+      hasNewUnsearchedFields: true,
+    });
+    const wrapper = shallow(<SchemaCallouts />);
+
+    expect(wrapper.find(UnsearchedFieldsCallout)).toHaveLength(1);
+  });
+
+  it('renders an unconfirmed fields callout if the schema has unconfirmed fields', () => {
+    setMockValues({
+      ...values,
+      hasUnconfirmedFields: true,
+    });
+    const wrapper = shallow(<SchemaCallouts />);
+
+    expect(wrapper.find(UnconfirmedFieldsCallout)).toHaveLength(1);
+  });
+
+  describe('UnsearchedFieldsCallout', () => {
+    it('renders an info callout about unsearched fields with a link to the relevance tuning page', () => {
+      const wrapper = shallow(<UnsearchedFieldsCallout />);
+
+      expect(wrapper.prop('title')).toEqual(
+        'Recently added fields are not being searched by default'
+      );
+      expect(wrapper.find('[data-test-subj="relevanceTuningButtonLink"]').prop('to')).toEqual(
+        '/engines/some-engine/relevance_tuning'
+      );
+    });
+  });
+
+  describe('UnconfirmedFieldsCallout', () => {
+    it('renders an info callout about unconfirmed fields', () => {
+      const wrapper = shallow(<UnconfirmedFieldsCallout />);
+
+      expect(wrapper.prop('title')).toEqual("You've recently added new schema fields");
+    });
+  });
+
+  describe('ConfirmSchemaButton', () => {
+    const actions = { updateSchema: jest.fn() };
```

Comment:
FYI: This button is new from ent-search. The current standalone button suggests that users manually click the "Update types" button in the top right hand corner of the page actions and does not provide a callout action/button:

### Before:
<img width="1071" alt="" src="https://user-images.githubusercontent.com/549407/117899651-69769500-b27c-11eb-970a-b0987f5dd827.png">

### After/Kibana:
<img width="1134" alt="" src="https://user-images.githubusercontent.com/549407/117899820-cb36ff00-b27c-11eb-86f4-c485a446cfa5.png">

IMO: This approach is better because:
- The action follows the message immediately. The UX flow doesn't take you jumping to separate places on the page.
- There's less conditional logic all over the page to change the main "Update types" button to "Confirm types" as well as enabling it. It's more predictable and less convoluted.

That being said - if you disagree, feel free to chime in here!